middleware/withShop.js

```diff
@@ -1,8 +1,16 @@
 module.exports = function withShop({ authBaseUrl } = {}) {
   return function verifyRequest(request, response, next) {
-    const { query: { shop }, session, baseUrl } = request;
+    const { session, baseUrl } = request;
+    let shop = request.query.shop;
```

Comment:
to be a bit more idiomatic of the code style favouring immutability we have in this repo I'd maybe consider something a bit more like 

```
const {session = {}, baseUrl, query} = request;
const {accessToken} = session;

const shop = shopFromReferrer(request.get('referrer')) || session.shop

if (accessToken) { 
  next();
  return;
}

if (shop) {
  response.redirect(`${authBaseUrl || baseUrl}/auth?shop=${shop}`);
  return;
}

response.redirect('/install');
```

Where we shove the regex logic into another function in scope.